### DISCUSSION TITLE
Job running while Ghostty is in not on the foreground is not exiting

### DISCUSSION THREAD

**Discussion by remosu** [ghostty-org/ghostty#2453] [Q&A] [answered]
Job running while Ghostty is in not on the foreground is not exiting.

- Execute a "long" running command and open another window. 
- Wait for it to end.
- Go back to Ghostty, the job is shown as  still running when it actually finished.

Ghostty on the left, iTerm2 on the right.

![Image](https://github.com/user-attachments/assets/e5c02480-ab30-4973-9533-a1106634beed)

After the job finish in Ghostty I need to `ctrl+c` it to be able to run another command. 

Only happening with `fish`. Tested the same thing in `zsh` and it works as expected.

Version: b38c5876
Build: 7726

**Comment by mitchellh** [comment, marked as answer]
I converted this to a discussion since I don't think this is going to be a bug. If it ends up being a bug we can convert it back to an issue. 

I'm a heavy fish user and this does not happen for me. Do you have any fish plugins? Can you try this with no fish config? 

For example there was this issue awhile back: https://github.com/ghostty-org/ghostty/issues/1917

Which we discovered was an upstream issue: https://github.com/julienXX/terminal-notifier/issues/301

**Reply by remosu** [reply]
Yep, that was it. 

I was using `franciscolourenco/done`. Removed it, and now the terminal doesn't hang. 

I'll keep an eye on the upstream issue.

Thanks!

_(I need to improve my search-fu. No idea how I missed #1917. 🤪)_
